### PR TITLE
Rewrite ListViewMAUI tests

### LINKED ISSUE #3617: MAUI tests fail frequently and aren't easily diagnosable
These are an attempt to shim in the old internal tests (note that MAUI here is an old internal test framework). The old tests worked quite differently and don't fit in well with the current test infrastructure. We're getting [frequent intermittent failures](https://dev.azure.com/dnceng/public/_build/results?buildId=737706&view=ms.vss-test-web.build-test-results-tab&runId=22844842&resultId=100000&paneView=history) like so:

```
System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(scenarioName: \"Hotkey_Fires_OnClick\")

Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Pass\r\nActual:   Fail\r\n          ↑ (pos 0)

Stack trace
   at System.Windows.Forms.Maui.IntegrationTests.MauiTestHelper.ValidateScenarioPassed(String projectName, String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/MauiTestHelper.cs:line 90
   at System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/WinformsMauiButtonTests.cs:line 35
``` 

This doesn't really help you much and it isn't obvious as to what you can do to move forward. In addition, the MAUI infrastructure sources aren't available so it is difficult to understand how the whole system works.

My recommendation (and current POR):

1. Disable the MAUI tests for now
2. Make them more xUnit-like and have any shim code in the repo
3. If we can make the tests plug in as a "normal" xUnit test then re-enable them for the CI, otherwise leave them for manual runs

cc: @AdamYoblick, @RussKie 

Failed with #3600

### PR DESCRIPTION
**Pull request by Danil-Andrianov**
Fixes #3617 

## Proposed changes

- Replaced ListViewMAUI tests into UI IntegrationTests project

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6257)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 1]
```diff
@@ -0,0 +1,1025 @@
...
+            });
+        }
+
+        [WinFormsFact]
+        public async Task ListView_Tile_FragmentNavigate_LastChild_ReturnsSubItem_ColumnsMoreThanSubItemsAsync()
+        {
+            await RunTestAsync((form, listView) =>
+            {
+                InitializeItems(listView, View.Tile, virtualModeEnabled: false, checkBoxesEnabled: false);
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:253]: nit: do not separate initialization calls with the blank line

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 2]
```diff
@@ -0,0 +1,1025 @@
...
+                Application.DoEvents();
+
+                AccessibleObject accessibleObject1 = listView.Items[0].SubItems[1].AccessibilityObject;
+                AccessibleObject accessibleObject2 = (AccessibleObject)accessibleObject1.FragmentNavigate(NavigateDirection.NextSibling)!;
+                AccessibleObject accessibleObject3 = (AccessibleObject)accessibleObject2.FragmentNavigate(NavigateDirection.NextSibling)!;
+                AccessibleObject accessibleObject4 = (AccessibleObject)accessibleObject3.FragmentNavigate(NavigateDirection.NextSibling)!;
+
+                Assert.Equal(accessibleObject2, listView.Items[0].SubItems[2].AccessibilityObject);
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:358]: nit: do not separate validations with a blank line, but add a blank line after validations are done

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 3]
```diff
@@ -0,0 +1,1025 @@
...
+                AccessibleObject expectedAccessibleItem2 = listView.Items[0].SubItems[2].AccessibilityObject;
+                AccessibleObject actualAccessibleItem1 = HitTest(GetSubItemLocation(0, 1));
+                AccessibleObject actualAccessibleItem2 = HitTest(GetSubItemLocation(0, 2));
+
+                Assert.Equal(actualAccessibleItem1, expectedAccessibleItem1);
+
+                Assert.Equal(actualAccessibleItem2, expectedAccessibleItem2);
+
+                AccessibleObject HitTest(Point point) =>
+                listView.AccessibilityObject.HitTest(point.X, point.Y)!;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:429]: nit: indent this line

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 4]
```diff
@@ -0,0 +1,1025 @@
...
+        private async Task RunTestAsync(Func<Form, ListView, Task> runTest)
+        {
+            await RunSingleControlTestAsync(
+                testDriverAsync: runTest,
+                createControl: () =>
+                {
+                    ListView control = new()
+                    {
+                        Location = new Point(0, 0),
+                        Size = new Size(439, 103)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:1011]: How did you determine sizes? Just copied from the old tests?

**Review by Tanya-Solyanik — COMMENTED** [review]
I'm not approving because I'd like someone more familiar with these tests to review it. @vladimir-krestov, @SergeySmirnov-Akvelon, please take a look.

Questions 
- most tests do not have the any input, why are they not in the unit test project? 
- where are the forms disposed?

**Review by vladimir-krestov — DISMISSED** [review]
Looks good ✔️, rework Tanya's points.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 5]
```diff
@@ -0,0 +1,1025 @@
...
+
+using System.Drawing;
+using WindowsInput.Native;
+using Xunit;
+using Xunit.Abstractions;
+using static Interop.UiaCore;
+
+namespace System.Windows.Forms.UITests
+{
+    public class ListViewTests : ControlTestBase
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:13]: There are a lot of duplicated tests that only differ on input parameters. Such tests must be converted into "theory" tests. Please work with @SergeySmirnov-Akvelon to reconcile those.

Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:13]: There are a lot of tests that look like "unit" tests, i.e. they don't require user input (i.e. keyboard/mouse). If there is a reason for them to be here, the reason must be noted in comments.

**Review by RussKie — CHANGES_REQUESTED** [review]

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs (hunk 4 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:1011]: You're right: there is no necessity in such accuracy. Nevertheless, we can't omit `Size` initialization: it causes tests fail. I could suggest round the values, e.g. (400, 100). What do you think about it?

Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:1011]: Sure

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs (hunk 5 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:13]: Now duplicated tests are united

**Comment by Danil-Andrianov** [comment]
@Tanya-Solyanik @RussKie 
Definitely the point about moving some tests to unit tests project not as simple as it appears at the first sight. I left commentary in the code about that point. Please, see the screen.
image

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+using WindowsInput.Native;
+using Xunit;
+using Xunit.Abstractions;
+using static Interop.UiaCore;
+
+namespace System.Windows.Forms.UITests
+{
+    // This class contains many tests which don't require user input.
+    // Although it looks like they should be moved to the unit tests project, it doesn't.
+    // Unit tests don't support the "Tile" mode, so all those tests belong here.
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:15]:
```suggestion
    // NOTE: This class contains many tests which don't require user input. Although they arguably belong to the unit
    // tests project, these tests assert behaviours of ListView.View=View.Tile, which doesn't work correctly unless
    // we ran an app.
```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+            {
+                InitializeItems(listView, View.Details, virtualModeEnabled, checkBoxesEnabled);
+                Point itemCenter = GetCenter(listView.RectangleToScreen(listView.Items[0].Bounds));
+
+                await MoveMouseAsync(form, itemCenter);
+
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Mouse.LeftButtonDoubleClick());
+                Assert.Equal(expected, listView.Items[0].Checked);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:40]: A typical test consists of three "A"s:
1. Arrange
2. Act
3. Assert

And it is helpful when these three "A"s are visually separated:
```suggestion
                InitializeItems(listView, View.Details, virtualModeEnabled, checkBoxesEnabled);

                Point itemCenter = GetCenter(listView.RectangleToScreen(listView.Items[0].Bounds));
                await MoveMouseAsync(form, itemCenter);
                await InputSimulator.SendAsync(
                    form,
                    inputSimulator => inputSimulator.Mouse.LeftButtonDoubleClick());

                Assert.Equal(expected, listView.Items[0].Checked);
```

Ditto below

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 8]
```diff
@@ -0,0 +1,511 @@
...
+        [InlineData(4, 3, 150, 150, 0, 3, (int)NavigateDirection.LastChild)]
+        [InlineData(4, 1, 150, 150, 0, 1, (int)NavigateDirection.LastChild)]
+        [InlineData(2, 5, 150, 150, 0, 1, (int)NavigateDirection.LastChild)]
+        [InlineData(10, 10, 100, 100, 0, 5, (int)NavigateDirection.LastChild)]
+        public async Task ListView_Tile_FragmentNavigate_WorksExpectedAsync(int columnCount, int subItemsCount, int width, int height, int itemIndex, int subItemIndex, int direction)
+        {
+            await RunTestAsync((form, listView) =>
+            {
+                InitializeTileList(listView, columnCount, subItemsCount, tileSize: new Size(width, height));
+                Application.DoEvents();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:55]: Why do we need this?

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 9]
```diff
@@ -0,0 +1,511 @@
...
+                Assert.Equal(-1, listView.Items[0].SubItems[1].AccessibilityObject.Column);
+                Assert.Equal(-1, listView.Items[0].SubItems[2].AccessibilityObject.Column);
+                Assert.Equal(-1, listView.Items[0].SubItems[2].AccessibilityObject.Column);
+
+                return Task.CompletedTask;
+            });
+        }
+
+        [WinFormsFact]
+        public async Task Click_On_Second_Column_Does_Not_Alter_CheckBoxesAsync()
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:384]: [naming] https://github.com/dotnet/winforms/blob/main/docs/testing.md#naming

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 10]
```diff
@@ -0,0 +1,511 @@
...
+                InitializeItems(listView, View.Details, virtualModeEnabled: false, checkBoxesEnabled: true);
+                foreach (ListViewItem item in listView.Items)
+                {
+                    Assert.Equal(0, item.StateImageIndex);
+                }
+
+                foreach (ListViewItem item in listView.Items)
+                {
+                    Assert.False(item.Selected);
+                }
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:397]: Any reason why these couldn't be joined?

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 11]
```diff
@@ -0,0 +1,511 @@
...
+
+                await MoveMouseAsync(form, listViewCenter);
+
+                await InputSimulator.SendAsync(
+                   form,
+                   inputSimulator => inputSimulator.Keyboard.KeyDown(VirtualKeyCode.SHIFT));
+
+                await InputSimulator.SendAsync(
+                    form,
+                    inputSimulator => inputSimulator.Mouse.LeftButtonClick());
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:409]: I suspect these can be amalgamated together:
```suggestion
                await InputSimulator.SendAsync(
                   form,
                   inputSimulator => 
                       inputSimulator.Keyboard.KeyDown(VirtualKeyCode.SHIFT))
                                     .Mouse.LeftButtonClick());
```

**Review by RussKie — COMMENTED** [review]
Looking better 👍

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs (hunk 6 above)
Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:15]: If you haven't installed https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelinesPreview, I strongly encourage to do so.
You'll need install "Developer Analytics tools" too, see https://github.com/pharring/EditorGuidelines/issues/87#issuecomment-965794495 for more details.

**Comment by RussKie** [comment]
> * where are the forms disposed?

Here
https://github.com/dotnet/winforms/blob/8ea3dfb805969d66fc0d96b5911c8a2d797d99f1/src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/Infra/ControlTestBase.cs#L235

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs (hunk 7 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:40]: @Tanya-Solyanik On what test style should I focus?
https://github.com/dotnet/winforms/pull/6257#discussion_r762255582

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs (hunk 8 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:55]: This method let us wait while all form's elements will drawn to interact with them correctly.

Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:56]: I'm still unconvinced, `InitializeTileList` executes synchronously and it isn't doing any long-running or CPU-intensive work.

**Comment by Danil-Andrianov** [comment]
> * where are the forms disposed?

Thanks for the question, now form can be disposed. Please, check my last commit.

**Review by RussKie — APPROVED** [review]

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs [hunk 12]
```diff
@@ -0,0 +1,521 @@
...
+                    {
+                        Size = new Size(400, 100)
+                    };
+
+                    return control;
+                },
+                createForm: () =>
+                {
+                    return new();
+                });
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ListViewTests.cs:518]: We probably want to be more explicit about the size of the form. E.g., what would happen if the test form was somehow narrower or shorter than (400, 100)? 
```suggestion
                createControl: () =>
                {
                    ListView control = new()
                    {
                       Dock = DockStyle.Fill
                    };

                    return control;
                },
                createForm: () =>
                {
                    return new()
                    {
                        Size = new Size(500, 400)
                    };
                });
```

**Review by RussKie — APPROVED** [review]

**Pull request merged**